### PR TITLE
[openwrt-18.06] python-cryptography: Add support for LibreSSL 2.7.x and fix CVE-2018-10903

### PR DESCRIPTION
**Pull request by BKPepe**
Maintainer: @commodo , @jefferyto 
Compile tested: Turris Omnia, mvebu (cortex-a9), OpenWrt 18.06.04
Run tested: Turris Omnia, mvebu (cortex-a9), OpenWrt 18.06.04

Description:

Due to the latest update of Python 2, the compilation of python-cryptography fails. Because of that, we need to backport commits from python-cryptography repository, which adds support for LibreSSL 2.7.x. On the other hand, we fixed [CVE-2018-10903](https://nvd.nist.gov/vuln/detail/CVE-2018-10903). 

To sum it up, there are added two patches and in both of them, they are included details, why they are needed.

Fixes https://github.com/openwrt/packages/issues/10329

### CODE REVIEW

**Comment by BKPepe** [comment]
@jefferyto , @commodo did you have time to check it, please? :)

File: lang/python/python-cryptography/patches/003-add-libressl-27x-support.patch [hunk 1]
```diff
@@ -0,0 +1,182 @@
+From d651daa2ffab54f0b7f9ab1177db7c58486bc785 Mon Sep 17 00:00:00 2001
+From: Paul Kehrer <paul.l.kehrer@gmail.com>
+Date: Thu, 31 May 2018 10:35:40 +0800
+Subject: [PATCH] LibreSSL 2.7.x support
+
+Pull request: https://github.com/pyca/cryptography/pull/4270
+Fixes issue: https://github.com/openwrt/packages/issues/10329
+introduced with Python 2.7.1.7
```
Comment (jefferyto) [inline lang/python/python-cryptography/patches/003-add-libressl-27x-support.patch]:
```suggestion
introduced with Python 2.7.17
```

Comment (BKPepe) [reply lang/python/python-cryptography/patches/003-add-libressl-27x-support.patch]: Fixed.
